### PR TITLE
MVP of a signal messenger test

### PR DESCRIPTION
**Pull request by hellais**
Basic implementation of a Signal app test

Reference issue: https://github.com/ooni/probe-engine/issues/278

### CODE REVIEW

File: internal/engine/experiment/signal/signal.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	urlgetter.RegisterExtensions(measurement)
+
+	certPool := netx.NewDefaultCertPool()
+	if certPool.AppendCertsFromPEM([]byte(signalCA)) == false {
+		return errors.New("AppendCertsFromPEM failed")
+	}
+
+	inputs := []urlgetter.MultiInput{
+		// Here we need to provide the method explicitly. See
+		// https://github.com/ooni/probe-cli/v3/internal/engine/issues/827.
```
Comment (bassosimone) [inline internal/engine/experiment/signal/signal.go:114]: LOL, this is a URL that should not have changed when I integrated probe-cli into probe-engine :facepalm: :grimacing:

File: internal/engine/experiment/signal/signal_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+func TestNewExperimentMeasurer(t *testing.T) {
+	measurer := signal.NewExperimentMeasurer(signal.Config{})
+	if measurer.ExperimentName() != "signal" {
+		t.Fatal("unexpected name")
+	}
+	if measurer.ExperimentVersion() != "0.2.0" {
+		t.Fatal("unexpected version")
+	}
+}
```
Comment (bassosimone) [inline internal/engine/experiment/signal/signal_test.go:21]:
```suggestion
}

```

**Comment by hellais** [comment]
The failing tests are unrelated to this PR.

File: internal/engine/experiment/signal/signal.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
+// Package signal contains the Signal network experiment.
+//
+// See https://github.com/ooni/spec/blob/master/nettests/ts-XXX-signal.md.
```
Comment (bassosimone) [inline internal/engine/experiment/signal/signal.go:3]:
```suggestion
// See https://github.com/ooni/spec/blob/master/nettests/ts-029-signal.md
```

**Pull request merged**